### PR TITLE
Migrate to Bzlmod

### PR DESCRIPTION
**Pull request by jadenPete**

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -1,6 +1,135 @@
...
+module(name = "rules_twirl")
+
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "buildifier_prebuilt", version = "7.3.1")
+bazel_dep(name = "rules_java", version = "7.11.1")
+bazel_dep(name = "rules_pkg", version = "1.0.1")
+bazel_dep(name = "rules_jvm_external", version = "6.4")
+bazel_dep(name = "rules_scala_annex")
+
+rules_scala_annex_version = "bzlmod-migration"
```
Comment (jadenPete) [inline MODULE.bazel]: - [ ] Wait until we've merged https://github.com/lucidsoftware/rules_scala/pull/74 and update this.

File: MODULE.bazel [hunk 2]
```diff
@@ -1,6 +1,137 @@
-###############################################################################
-# Bazel now uses Bzlmod by default to manage external dependencies.
-# Please consider migrating your external dependencies from WORKSPACE to MODULE.bazel.
-#
-# For more details, please check https://github.com/bazelbuild/bazel/issues/18958
-###############################################################################
+module(name = "rules_twirl")
+
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "buildifier_prebuilt", version = "7.3.1")
```
Comment (jjudd) [inline MODULE.bazel]: I think we can mark this as `dev_dependency`

File: MODULE.bazel [hunk 3]
```diff
@@ -1,6 +1,137 @@
...
+bazel_dep(name = "rules_java", version = "7.11.1")
+bazel_dep(name = "rules_pkg", version = "1.0.1")
```
Comment (jjudd) [inline MODULE.bazel]: I think we can mark this as `dev_dependency`

File: MODULE.bazel [hunk 4]
```diff
@@ -1,6 +1,137 @@
...
+rules_scala_annex_version = "bzlmod-migration"
+
+archive_override(
+    module_name = "rules_scala_annex",
+    integrity = "sha256-Gt+S2R53KPhfsK66bhOSXowjb35Ouf24zcT3xFZcyAs=",
+    strip_prefix = "rules_scala-{}".format(rules_scala_annex_version),
+    urls = ["https://github.com/lucidsoftware/rules_scala/archive/refs/heads/{}.zip".format(rules_scala_annex_version)],
+)
+
+bazel_dep(name = "stardoc", version = "0.7.1")
```
Comment (jjudd) [inline MODULE.bazel]: Can this be moved up with the other `bazel_deps`?

Also I think we can mark this as `dev_dependency`

File: MODULE.bazel [hunk 5]
```diff
@@ -1,6 +1,137 @@
...
+
+twirl_version = "2.0.7"
+
+protobuf_version = "4.28.3"
+
+scopt_version = "4.1.0"
+
+specs2_version = "4.20.8"
+
+zinc_version = "1.10.1"
```
Comment (jjudd) [inline MODULE.bazel]: Scope creep, but can we please update this to 1.10.4? Annex is already on that version, so it should be as simple as changing the version and updating the lockfile.

**Review by jjudd — COMMENTED** [review]
Mostly LGTM. Left a few comments.

File: MODULE.bazel (hunk 2 above)
Comment (jadenPete) [reply MODULE.bazel]: See my comment here:
https://github.com/lucidsoftware/rules_scala/pull/74#discussion_r1873510258

File: MODULE.bazel (hunk 3 above)
Comment (jadenPete) [reply MODULE.bazel]: See my comment here:
https://github.com/lucidsoftware/rules_scala/pull/74#discussion_r1873510258

File: MODULE.bazel (hunk 4 above)
Comment (jadenPete) [reply MODULE.bazel]: See my comment here:
https://github.com/lucidsoftware/rules_play_routes/pull/57#discussion_r1873547143

File: MODULE.bazel (hunk 5 above)
Comment (jadenPete) [reply MODULE.bazel]: Sure thing.

**Review by jjudd — APPROVED** [review]

**Comment by jadenPete** [comment]
Squashed.

**Pull request merged**